Part 1 - React tutorial: The beginner's guide to learning React in 2020.

lojacho opened the pull request:
In this pull request, I was able to do all the things that where described in the next [link](https://ibaslogic.com/react-tutorial-for-beginners/)

Commit:
Delete all files in src folder

Commit:
Create components files

Commit:
Render multiple html tag using a div

Commit:
Render multiple HTML tags using React.Fragment

Commit:
Activate checks, logs warning at runtime

Commit:
Access the state data from within the child compon

Commit:
Help identify which items will change, add, remove

Commit:
Linke header in the main container

Commit:
Addd checkbox to the todo items

Commit:
Make the input ckeckbox a controlled input

Commit:
Add an onChange handler to keep track of changes

Commit:
Start communication from the parent component

GutemaG reviewed (changes requested):
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @dgcuenca 


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 
## Highlights
✅ Great use of GitHub flow. 💯 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️

You did a great job working on the React project! 🚀 However, I noticed that your project  is lacking some information:

- [ ] Please Add react/redux linter and make sure there is no error
- [ ] Please Consider adding microverse README template and MIT license.
- [ ] There are some components are empty so try to implement them.

Kindly consider applying these changes! 🛠️ 

Cheers and Happy coding!👏👏👏

Commit:
Pass method to TodosList through the props

Commit:
Pass prop to the todoItem component

Commit:
Enable comunication, access from child component

Commit:
Identify which checkbox is clicked

Commit:
Receive and logging the id

Commit:
Tell react we update the state using setState

Commit:
Guarantee receive UP TO DATE state version

Commit:
Quit strict-mode limitations refactor state update

Commit:
Use return statement as an alternative

previous method also works

Commit:
Add delete button

Commit:
Start enabling communications betwween components

Commit:
Construct the path for enable communication

Commit:
Finish enabling communication DELETE

Commit:
Remove any of the deleted items from the list

note the spread operator (…) in the code. It allows us to grab the current todos item(s) at every point. As this is necessary for the code to work.

Commit:
Add form

Commit:
Have a state to manage the user's input.

In order to make the form input field a controlled field

Commit:
Add state to input field value

Commit:
Enable take data from the input field

Commit:
Allow handle multiple input fields

Commit:
Prevent default behaviour of the form submission

Commit:
Enable communication for adding new task

Commit:
Generate random ids for the todos list items

Commit:
Validate the Form Submission

Commit:
Style with CSS stylesheet

Commit:
Apply inline styling

Valid JavaScript expressions in JSX are written inside curly braces. The second curly brace is for the inline styling in the form of a JavaScript object.
Style keys are in camelCase.

Commit:
Style inline using variables

Commit:
Style with CSS Modules

The styles applied will be scoped only to that component. This way, you can use the same class names in different components without worrying about conflicts with the CSS selector.

Commit:
Style when any of the todos items is completed

Commit:
Using Destructuring for optimize code

Commit:
Add a reference handler when clickX2 a todoItem

Commit:
See the edit field on double click

Commit:
Convert to a controlled field

Commit:
Include an onChange event in order to control it

Commit:
Enable path communication

Commit:
Can edit and update the todos title

Commit:
Detect when user press Enter to submit edited Item

Commit:
Delete hardcoded todos data

Commit:
Fetch data from a remote endpoint

Commit:
Save the todos (if exist) in the local storage.

Commit:
Get the stored items and add back to the view

Commit:
Watch in action componentWillUnmount
